Lombiq.NodeJs.Extensions/package.json

```diff
@@ -3,55 +3,78 @@
+      "TARGET_DIR=$(node scripts/get-target) SCSS_SOURCE=$npm_config_source SCSS_TARGET=$npm_config_target pnpm _compile:styles",
     "_compile:styles":
-      "run-s _sass:styles _autoprefix:styles _minify:styles _rm:temp:styles || ERRORSTEP=compilation pnpm _error:styles",
+      "run-s --print-label _sass:styles _autoprefix:styles _minify:styles _rm:temp:styles || AREA=styles STEP=compilation pnpm _error:styles",
     "_sass:styles": "cd $TARGET_DIR && sass $SCSS_SOURCE:$SCSS_TARGET/_temp_ --embed-sources",
     "_autoprefix:styles":
       "cd $TARGET_DIR && postcss $SCSS_TARGET/_temp_/**/*.css --base $SCSS_TARGET/_temp_ --dir $SCSS_TARGET --use autoprefixer --map",
     "_minify:styles":
       "cd $TARGET_DIR && cleancss --batch --batch-suffix .min \"$SCSS_TARGET/**/*.css\" \"!$SCSS_TARGET/**/*.min.css\" \"!$SCSS_TARGET/_temp_/**/*\"",
     "_rm:temp:styles": "cd $TARGET_DIR && rimraf $SCSS_TARGET/_temp_",
-    "clean:styles": "TARGET_DIR=../../.. SCSS_TARGET=wwwroot/css pnpm _clean:styles",
-    "clean:styles:args": "TARGET_DIR=../../.. SCSS_TARGET=$npm_config_target pnpm _clean:styles",
+    "clean:styles": "TARGET_DIR=$(node scripts/get-target) SCSS_TARGET=wwwroot/css pnpm _clean:styles",
+    "clean:styles:args": "TARGET_DIR=$(node scripts/get-target) SCSS_TARGET=$npm_config_target pnpm _clean:styles",
     "_clean:styles": "cd $TARGET_DIR && rimraf \"$SCSS_TARGET/**/*.*\"",
     "watch:styles": "SCSS_SOURCE=Assets/Styles SCSS_TARGET=wwwroot/css pnpm _watch:styles",
     "watch:styles:args": "SCSS_SOURCE=$npm_config_source SCSS_TARGET=$npm_config_target pnpm _watch:styles",
     "_watch:styles":
       "nodemon --watch ../../$SCSS_SOURCE --ext scss --exec \"pnpm build:styles:args --source=$SCSS_SOURCE --target=$SCSS_TARGET\"",
-    "_error:styles": "echo \"Error in styles pipeline: $ERRORSTEP failed.\"",
-    "--": "-----------------------------------------------------------------------------------------------------------",
-    "__shell_docs__": "https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-shell/tests/shell.test.ts"
+    "---": "------------------------------------------------- SCRIPTS ------------------------------------------------",
+    "build:scripts": "run-p --print-label --continue-on-error lint:scripts minify:scripts",
+    "lint:scripts":
+      "(TARGET_DIR=$(node scripts/get-target) && node scripts/exists $TARGET_DIR/Assets/Scripts || exit 0 && cd $TARGET_DIR npx eslint Assets/Scripts --max-warnings 0 --config ${INIT_CWD}/config/eslint.config.json) || AREA=scripts STEP=linting pnpm _error",
+    "minify:scripts": "run-s --print-label compile:scripts _minify:scripts",
+    "compile:scripts":
+      "(TARGET_DIR=$(node scripts/get-target) && node scripts/exists $TARGET_DIR/Assets/Scripts || exit 0 && cd $TARGET_DIR && babel Assets/Scripts --out-dir wwwroot/js --config-file ${INIT_CWD}/config/babel.config.json) || AREA=scripts STEP=compiling pnpm _error ",
```

Comment:
These lines are really long, break them up a bit.

Reply:
I've shortened them, but there's no way to break them up. It'd be possible to defer parts of the call chain to another script entry, but every invocation comes with a not-so-small cost, so I'm trying to avoid unnecessary calls.

Reply:
Why is there no way to break them?

Reply:
I see, of course.

Reply:
In this longest line of all (showing the current version) I'm still considering two things:
```json
    "compile:scripts":
      "SRC=$(node scripts/get-path js source) DEST=$(node scripts/get-path js target) && babel $SRC --out-dir $DEST --config-file ${INIT_CWD}/config/babel.config.json || AREA=scripts STEP=compilation pnpm error",
```

1. Renaming the `scripts` folder to `_` which would be shorter, somehow OK due to `underscore.js` using the same, and those scripts being internal, anyways. That would save **12 chars** here.
2. Moving _babel.config.json_ into the parent folder, which would be `Lombiq.NodeJs.Extensions`' root folder. Then, I wouldn't have to pass the path to the config explicitly.
3. Moving _babel.config.json_  into the consumer project, just like _.stylelintrc_ and _.eslintrc_. This would have the same effect as 2. with the addition that it would allow to easily opt out of transpilation to ES2015 by removing parts of its content. I've wanted to do this initially, but then deemed it out of scope.

What do you think?

Reply:
I wouldn't do 1 since it hurts understandability. 3 sounds good instead of 2.

Reply:
I'm refraining from changing this right now because the babel config file also contains settings for the sitemap generation and I wouldn't like that to be exposed to the consumer. So I'm leaving this part as is.